src/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.X509.cs

```diff
@@ -117,26 +125,50 @@ internal static byte[] X509GetRawData(SafeSecCertificateHandle cert)
             bool exportable,
             out SafeSecIdentityHandle identityHandle)
         {
-            SafeSecCertificateHandle certHandle;
-            int osStatus;
-            int ret;
-
-            SafeCreateHandle cfPassphrase = s_nullExportString;
+            SafeCreateHandle cfPassphrase = null;
             bool releasePassword = false;
 
             try
             {
                 if (!importPassword.IsInvalid)
                 {
                     importPassword.DangerousAddRef(ref releasePassword);
-                    IntPtr passwordHandle = importPassword.DangerousGetHandle();
+                    cfPassphrase = CoreFoundation.CFStringCreateFromSpan(importPassword.DangerousGetSpan());
+                }
 
-                    if (passwordHandle != IntPtr.Zero)
-                    {
-                        cfPassphrase = CoreFoundation.CFStringCreateWithCString(passwordHandle);
-                    }
+                return X509ImportCertificate(
+                    bytes,
+                    contentType,
+                    cfPassphrase,
+                    keychain,
+                    exportable,
+                    out identityHandle);
```

Comment:
should identityHandle be freed? (not clear about the ownership)

Reply:
(also you seem to be disposing it below)

Reply:
`identityHandle` is an out/return value... I'm not sure why we'd want to dispose it...

Reply:
ahh, missed the `out` on the signature, thought this was unused